packages/opencensus-web-instrumentation-zone/test/test-perf-resource-timing-selector.ts

```diff
@@ -0,0 +1,216 @@
+/**
+ * Copyright 2019, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Span } from '@opencensus/web-core';
+import {
+  getPerfResourceEntries,
+  getPossiblePerfResourceEntries,
+  getXhrPerfomanceData,
+} from '../src/perf-resource-timing-selector';
+import { spyPerfEntryByType, createFakePerfResourceEntry } from './util';
+
+describe('Perf Resource Timing Selector', () => {
+  describe('getPerfResourceEntries', () => {
+    it('Should filter properly the resource entries', () => {
+      // Should ignore this as it has a different name
+      const entry1 = createFakePerfResourceEntry(
+        13.100000011036173,
+        15.100000011036173,
+        '/different_url'
+      );
+      const entry2 = createFakePerfResourceEntry(
+        14.100000011036173,
+        15.200000011036173
+      );
+      const entry3 = createFakePerfResourceEntry(
+        16.100000011036173,
+        18.100000011036173
+      );
+      // Should ignore this even if the end times are close.
+      const entry4 = createFakePerfResourceEntry(
+        20.100000011036173,
+        30.000000000000003
+      );
+      // Ignore this as its out of the span end time.
+      const entry5 = createFakePerfResourceEntry(
+        35.100000011036173,
+        40.100000011036173
+      );
+      const perfResourceEntries = [entry1, entry2, entry3, entry4, entry5];
+      spyPerfEntryByType(perfResourceEntries);
+      const span = createSpan(12, 30);
+
+      const entriesToBeFiltered = [entry2, entry3];
+      const filteredPerfEntries = getPerfResourceEntries('/test', span);
+      expect(filteredPerfEntries).toEqual(entriesToBeFiltered);
+    });
+  });
+
+  describe('getPossiblePerfResourceEntries', () => {
+    it('Should pair no overlapping entries', () => {
+      const entry1 = createFakePerfResourceEntry(
```

Comment:
What would you think about calling these something like `overlappingEntry1` and `nonOverlappingEntry2`, etc. to indicate a bit more what the fake data is showing?

Reply:
As part of taking the XhrPerformaceResourceTiming type as an object, instead of labeling these objects like that, I decided to create the expected (non)overlapping entries following.